src/mtev_reverse_socket.c

```diff
@@ -387,6 +412,7 @@ mtev_reverse_socket_channel_handler(eventer_t e, int mask, void *closure,
     eventer_remove_fd(e->fd);
     CHANNEL.pair[0] = CHANNEL.pair[1] = -1;
     mtev_reverse_socket_channel_shutdown(cct->parent, cct->channel_id, e);
+    mtev_reverse_socket_deref(cct->parent);
```

Comment:
So the channl_shutdown right before this can possibly deref as well... Is a double deref in that case intended?